include/sddf/serial/queue.h

```diff
@@ -198,13 +191,13 @@ static inline uint32_t serial_queue_free(serial_queue_handle_t *queue_handle)
 
 static inline uint32_t serial_queue_contiguous_free(serial_queue_handle_t *queue_handle)
 {
-    uint32_t tail = queue_handle->queue->tail;
+    uint32_t tail = queue_handle->queue->tail % queue_handle->size;
     uint32_t free_size = serial_queue_free(queue_handle);
 
-    if ((tail + free_size) <= UINT32_MAX || (tail + free_size == 0)) {
+    if ((tail + free_size) <= queue_handle->size && (tail + free_size > tail)) {
         return free_size;
     } else {
```

Comment:
Get rid of the else as there's an early return.

Reply:
Refactored this function